packages/nx/src/utils/lock-file/npm.ts

```diff
@@ -402,11 +442,199 @@ export function transitiveDependencyNpmLookup(
  */
 export function pruneNpmLockFile(
   lockFileData: LockFileData,
-  packages: string[]
+  packages: string[],
+  projectName?: string
 ): LockFileData {
-  // todo(meeroslav): This functionality has not been implemented yet
-  console.warn(
-    'Pruning package-lock.json is not yet implemented. Returning entire lock file'
-  );
-  return lockFileData;
+  // NPM V1 does not track full dependency list in the lock file,
+  // so we can't reuse the lock file to generate a new one
+  if (lockFileData.lockFileMetadata.metadata.lockfileVersion === 1) {
+    console.warn(
```

Comment:
NPM V1 lock file only contains information about required and nested dependencies, but not peer dependencies so we cannot rebuild the entire pruned lock file accurately.